lib/src/component/abstract_transition.dart

```diff
@@ -155,27 +168,15 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps,
     prepareHide();
 
     setState(newState()
-      ..transitionPhase = hasTransition ? TransitionPhase.HIDING : TransitionPhase.HIDDEN
+      ..transitionPhase = hasTransitionOut ? TransitionPhase.HIDING : TransitionPhase.HIDDEN
     );
   }
 
   /// Listens for the next `transitionend` event and invokes a callback after
   /// the event is dispatched.
   @mustCallSuper
   void onNextTransitionEnd(complete()) {
-    var skipCount = props.transitionCount - 1;
-
-    if (props.transitionCount <= 0) {
-      var warningMessage = 'You have set `props.transitionCount` to an invalid option: ${props.transitionCount}.';
-
-      if (props.transitionCount == 0) {
-        warningMessage += ' Instead of setting this prop to 0, override the `hasTransition` getter to return false.';
-      }
```

Comment:
These are no longer necessary; transition counts that are `<=0` are accounted for in `hasTransitionIn` and `hasTransitionOut`, and behave as if there is no transition (see tests).